open-apparel-registry/open-apparel-registry
# Delay loading search field dropdown data

## Overview

As part of our redesign of the sidebar to support extended fields we researched search behavior and hid list frequently used search fields. If the search fields are never shown, their dropdown list option do not need to be fetched.

Connects #1729

## Demo

![2022-03-19 10 04 11](https://user-images.githubusercontent.com/17363/159131051-b05297d0-c8f7-48a7-8e90-5ee7bdf66a21.gif)

## Notes

When refreshing the page after selection both a facility type and a processing type, the facility type options do not populate. This is not a regression and is a non-ideal behavior in the current implementation of filtering options based on the connection between facility type and processing type.  

## Testing Instructions

* Open the dev tools network tab and filter by XHR
* Browse http://localhost:6543/ and verify that API request are not made for
  * /api/contributor-types
  * /api/facility-processing-types
  * /api/product-types
  * /api/worker-ranges
* Click "More filters" and verify that fetches are now make and that all the dropdowns are populated
* Select items in all the extended field dropdowns then refresh the page, verify that the selections are restored

## Checklist
